### PR TITLE
dmidecode version bump to 3.2.8

### PR DESCRIPTION
**Pull request by superherointj**
###### Motivation for this change
[PR#97614](https://github.com/NixOS/nixpkgs/pull/97614) was missing dmidecode's version bump.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ajs124** [comment]
Shouldn't this be setting `pname` and `version` nowadays?

**Review by samueldr — CHANGES_REQUESTED** [review]
Quoting @blitz 

> This is broken, because it tries to fetch dmidecode-3.2.8.tar.xz from the savannah mirrors, which does not exist. See http://ftp.twaren.net/Unix/NonGNU/dmidecode/

https://github.com/NixOS/nixpkgs/pull/98175#pullrequestreview-490908495

**Comment by samueldr** [comment]
Additionally, this is not truly "3.2.8", upstream still refers to it as 3.2...

We don't have a way to differentiate between versions of unversioned patched-up software defined and/or documented. Nothing similar to what I believe debian does, adding "+n" where n is the amount of debian-specific updates the package had.

This is causing only an issue with `nix-env`, as far as I can tell. Thus, I don't know what is the best way to approach this problem.

**Comment by blitz** [comment]
> Additionally, this is not truly "3.2.8", upstream still refers to it as 3.2...

Yes, it looks like there is a 3.2.8 release when there really isn't.
>
> We don't have a way to differentiate between versions of unversioned patched-up software defined and/or documented. Nothing similar to what I believe debian does, adding "+n" where n is the amount of debian-specific updates the package had.

Fedora seems to call it `3.2-8`: https://pkgs.org/search/?q=dmidecode

File: pkgs/os-specific/linux/dmidecode/default.nix [hunk 1]
```diff
@@ -1,13 +1,14 @@
 { stdenv, fetchurl, fetchpatch }:
 
 stdenv.mkDerivation rec {
-  name = "dmidecode-3.2";
-
+  pname = "dmidecode";
+  version = "3.2.8";
   src = fetchurl {
-    url = "mirror://savannah/dmidecode/${name}.tar.xz";
+    # Download version in upstream is 3.2. Version 3.2.8 does not come from upstream and was made up to differentiate the applied recommended patches bellow.
```
Comment (blitz) [inline pkgs/os-specific/linux/dmidecode/default.nix:7]:
```suggestion
    # Download version in upstream is 3.2. Version 3.2.8 does not come from upstream
    # and was made up to differentiate the applied recommended patches below.
```

File: pkgs/os-specific/linux/dmidecode/default.nix [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
+    url = "mirror://savannah/dmidecode/${pname}-3.2.tar.xz";
     sha256 = "1pcfhcgs2ifdjwp7amnsr3lq95pgxpr150bjhdinvl505px0cw07";
   };
-
+
```
Comment (blitz) [inline pkgs/os-specific/linux/dmidecode/default.nix]: This line introduces a whitespace error. :space_invader:

**Comment by superherointj** [comment]
Which sort of version to use then "3.2-8" or "3.2.8"? Arch would do "3.2-1". The "1" refers to overwriting the standard package (and not number of patches applied).

**Comment by AndersonTorres** [comment]
As I have seen here, there is no need for version bumping. It is just a convention, and in Nixpkgs those things have way less importance compared with other distros.

By updating the expression with the required patches, the build bots will update the package automatically (after a couple of days I suppose).

**Comment by AndersonTorres** [comment]
On the other hand, I suggest creating a new expression, something like "dmidecode/unstable.nix" fetching the last Giot sources. But it can wait...

**Pull request closed without merging**